libs/shared/components/src/auth/UserMenu/UserMenu.tsx

```diff
@@ -14,17 +26,26 @@ export const UserMenu = ({ fullscreen = false }: { fullscreen?: boolean }) => {
   const handleClick = () => {
     setDropdownState(dropdownState === 'open' ? 'closed' : 'open')
   }
+  useEffect(() => {
+    setUserMenuOpen && setUserMenuOpen(dropdownState === 'open' ? true : false)
+  }, [dropdownState])
+
+  useEffect(() => {
+    if ((dropdownState === 'open') !== userMenuOpen) {
+      setDropdownState('closed')
+    }
+  }, [userMenuOpen])
```

Comment:
I find this a bit confusing. if `dropdownState = closed` and `userMenuOpen = true` then we will get `false !== true` and set the dropdown state to `closed` but it is already closed. Shouldn't it be changed to open?
And also why have two variables keeping the open/closed state instead of just exposing the dropdownState?

Reply:
Yes it is confusing! 🤯  took me a while to understand it myself 😆 

"userMenuOpen" and "setUserMenuOpen" is to control the dropdown state as well. But I have to access it through the Header component. 
The new design has this language button and X button that is displayed when a user has the "UserMenu" open, but only in service portal. 

This check is to make sure that the X button (that is displayed in the Header component) is closing the UserMenu when the dropdownState is open.
Basically if I press X then the state changes and I check if the dropdown is open, if so I close it. 
I changed the code to this - hopefully more readable? : 

`useEffect(() => {
    if (dropdownState === 'open' && !userMenuOpen) {
      setDropdownState('closed')
    }
  }, [userMenuOpen])
`

This is all getting quite bloated & confusing and it is time to refactor it all in the next refactor week! I also wonder if it is better to use separate component in the Service Portal 🤔

Reply:
I see and yes this is more readable.